File: scratch/migrate.js
const fs = require('fs');
const path = require('path');

const rootDir = 'E:\\EIMG\\EIMG Web\\EIMG Website';

function getFiles(dir, fileList = []) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const stat = fs.statSync(dir + '/' + file);
    if (stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.git' && file !== 'scratch') {
        getFiles(dir + '/' + file, fileList);
      }
    } else if (file.endsWith('.html')) {
      fileList.push(dir + '/' + file);
    }
  }
  return fileList;
}

const htmlFiles = getFiles(rootDir);
let results = [];

let enFilesRequiringTranslation = [];

for (const file of htmlFiles) {
  let content = fs.readFileSync(file, 'utf8');
  let originalContent = content;
  
  let dataI18nCount = (content.match(/\s+data-i18n="[^"]*"/g) || []).length;
  content = content.replace(/\s+data-i18n="[^"]*"/g, '');
  
  let scriptCount = (content.match(/<script\s+src="[^"]*i18n\.js"><\/script>\s*/g) || []).length;
  content = content.replace(/<script\s+src="[^"]*i18n\.js"><\/script>\s*/g, '');
  
  let sessionStorageScriptCount = (content.match(/<script>\s*\(\s*function\(\)\s*\{\s*sessionStorage\.setItem\('expertImmoLang','en'\);.*?\}\)\(\);\s*<\/script>\s*/g) || []).length;
  content = content.replace(/<script>\s*\(\s*function\(\)\s*\{\s*sessionStorage\.setItem\('expertImmoLang','en'\);.*?\}\)\(\);\s*<\/script>\s*/g, '');

  let langBtnCount = (content.match(/class="lang-btn/g) || []).length;
  
  // Replace the .lang-switcher styles
  content = content.replace(/\.lang-switcher\s*\{[^}]*\}\s*\.lang-switcher\s+a\s*\{[^}]*\}\s*\.lang-switcher\s+a\.active\s*\{[^}]*\}\s*/g, '');

  // For the switcher placeholders
  const isEn = file.replace(/\\/g, '/').includes('/en/');
  const langCurrent = isEn ? 'en' : 'fr';
  const mobileMount = `<div class="lang-switcher-mount-mobile" data-lang-current="${langCurrent}"></div>`;
  const desktopMount = `<div class="lang-switcher-mount" data-lang-current="${langCurrent}"></div>`;

  let switcherMatchCount = 0;
  content = content.replace(/<div class="lang-switcher"[^>]*>[\s\S]*?<\/div>/g, (match) => {
    switcherMatchCount++;
    if (switcherMatchCount === 1) return mobileMount;
    if (switcherMatchCount === 2) return desktopMount;
    return match;
  });

  if (content !== originalContent) {
    fs.writeFileSync(file, content, 'utf8');
  }

  if (isEn) {
    enFilesRequiringTranslation.push(file.replace(/\\/g, '/').replace(rootDir.replace(/\\/g, '/') + '/', ''));
  }

  results.push({
    file: file.replace(/\\/g, '/').replace(rootDir.replace(/\\/g, '/') + '/', ''),
    dataI18n: dataI18nCount,
    scripts: scriptCount,
    langBtns: langBtnCount,
    switchersReplaced: switcherMatchCount
  });
}

const i18nPath = path.join(rootDir, 'js', 'i18n.js');
let i18nDeleted = false;
if (fs.existsSync(i18nPath)) {
  fs.unlinkSync(i18nPath);
  i18nDeleted = true;
}

console.log(JSON.stringify({ results, enFilesRequiringTranslation, i18nDeleted }, null, 2));

File: scratch/migrate.py
import os
import re
import json

root_dir = r"E:\EIMG\EIMG Web\EIMG Website"
html_files = []

for dirpath, dirnames, filenames in os.walk(root_dir):
    # Exclude directories
    dirnames[:] = [d for d in dirnames if d not in ['.git', 'node_modules', 'scratch']]
    for f in filenames:
        if f.endswith('.html'):
            html_files.append(os.path.join(dirpath, f))

results = []
en_files_requiring_translation = []

for filepath in html_files:
    with open(filepath, 'r', encoding='utf-8') as file:
        content = file.read()
    
    original_content = content
    
    # data-i18n
    data_i18n_pattern = r'\s*data-i18n="[^"]*"'
    data_i18n_matches = len(re.findall(data_i18n_pattern, content))
    content = re.sub(data_i18n_pattern, '', content)
    
    # script i18n
    script_pattern = r'<script\s+src="[^"]*i18n\.js"></script>\s*'
    script_matches = len(re.findall(script_pattern, content))
    content = re.sub(script_pattern, '', content)
    
    # session storage script in /en/ pages
    session_script_pattern = r'<script>\s*\(\s*function\(\)\s*\{\s*sessionStorage\.setItem\(\'expertImmoLang\',\'en\'\);.*?\}\)\(\);\s*</script>\s*'
    session_script_matches = len(re.findall(session_script_pattern, content))
    content = re.sub(session_script_pattern, '', content)
    
    # lang-btn counts
    lang_btn_pattern = r'class="lang-btn'
    lang_btn_matches = len(re.findall(lang_btn_pattern, content))
    
    # style block for lang-switcher
    style_pattern = r'\.lang-switcher\s*\{[^}]*\}\s*\.lang-switcher\s+a\s*\{[^}]*\}\s*\.lang-switcher\s+a\.active\s*\{[^}]*\}\s*'
    content = re.sub(style_pattern, '', content)
    
    # replace switchers
    rel_path = os.path.relpath(filepath, root_dir).replace('\\', '/')
    is_en = '/en/' in rel_path or rel_path.startswith('en/')
    lang_current = 'en' if is_en else 'fr'
    
    mobile_mount = f'<div class="lang-switcher-mount-mobile" data-lang-current="{lang_current}"></div>'
    desktop_mount = f'<div class="lang-switcher-mount" data-lang-current="{lang_current}"></div>'
    
    switcher_pattern = r'<div class="lang-switcher"[^>]*>[\s\S]*?</div>'
    
    def replacer(match):
        replacer.count += 1
        if replacer.count == 1:
            return mobile_mount
        elif replacer.count == 2:
            return desktop_mount
        return match.group(0)
    
    replacer.count = 0
    content = re.sub(switcher_pattern, replacer, content)
    
    if content != original_content:
        with open(filepath, 'w', encoding='utf-8') as file:
            file.write(content)
            
    if is_en:
        en_files_requiring_translation.append(rel_path)
        
    results.append({
        'file': rel_path,
        'dataI18n': data_i18n_matches,
        'scripts': script_matches,
        'langBtns': lang_btn_matches,
        'switchersReplaced': replacer.count
    })

i18n_path = os.path.join(root_dir, 'js', 'i18n.js')
i18n_deleted = False
if os.path.exists(i18n_path):
    os.remove(i18n_path)
    i18n_deleted = True

print(json.dumps({
    'results': results,
    'enFilesRequiringTranslation': en_files_requiring_translation,
    'i18nDeleted': i18n_deleted
}, indent=2))
